include/sys/zfs_ioctl.h

```diff
@@ -338,6 +339,11 @@ typedef struct zinject_record {
 #define	ZEVENT_SEEK_START	0
 #define	ZEVENT_SEEK_END		UINT64_MAX
 
+/* Note percentage use case has a higher resolution */
+#define	zi_percentage	zi_freq
+#define	ZI_PERCENTAGE_MIN	4294UL
```

Comment:
Overloading `zi_freq` with values over 4294  to imply a percentage is clever.  Maybe too clever.

What if instead we simply changed the expected value of `.zi_freq` to always represent a scaled percentage in the range 0..UINT_MAX.  That would break compatibility with older versions of `zinject` but for a developer tool I'm OK with that.  That way could use the existing `-f` option and the other failure locations which use zi_freq would benefit from the improved precision.